src/renderer/coremods/contextMenu/index.tsx

```diff
@@ -16,26 +16,36 @@ export const menuItems = {} as Record<
   Array<{ getItem: GetContextItem; sectionId: number; indexInSection: number }>
 >;
 
+type RepluggedContextItem = ContextItem & {
+  props: { replug?: boolean; id?: string };
+};
+
 /**
  * Converts data into a React element. Any elements or falsy value will be returned as is
  * @param raw The data to convert
  * @returns The converted item
  */
-function makeItem(raw: RawContextItem | ContextItem | undefined | void): ContextItem | undefined {
+function makeItem(
+  raw: RawContextItem | ContextItem | undefined | void,
+): RepluggedContextItem | undefined {
   // Occasionally React won't be loaded when this function is ran, so we don't return anything
   if (!React) return undefined;
 
   if (!raw) {
     // If something falsy is passed, let it through
     // Discord just skips over them too
-    return raw as ContextItem | undefined;
+    return raw as undefined;
   }
   if (React.isValidElement(raw)) {
     // We can't construct something that's already made
-    return raw as ContextItem | undefined;
+    return raw as RepluggedContextItem;
   }
 
   const { type, ...props } = raw;
+  // add in prop  for cleanup
+  props.replug = true;
```

Comment:
You may want to move this back in `_buildPatchedMenu`, updating menu checkboxes doesn't delete them.